client/state/selectors/get-user-setting.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Given a settingName, returns that setting if it exists or null
+ *
+ * @param  {Object} state Global state tree
+ * @param  {String} settingName - setting name
+ * @return {*} setting name value
+ */
+export default function getUserSetting( state, settingName ) {
+	const { settings, unsavedSettings } = state.userSettings;
+	let setting = null;
+
+	// If we haven't fetched settings, or if the setting doesn't exist return null
+	if ( settings && settings[ settingName ] !== undefined ) {
+		setting = unsavedSettings[ settingName ] !== undefined
+			? unsavedSettings[ settingName ]
+			: settings[ settingName ];
+	}
+
+	return setting;
```

Comment:
I'm a bit confused glossing over this nested conditional. I wonder if we could simplify the logic a bit…

```js
/**
 * Get user setting, prefer returning unsaved values first
 */
export const getUserSetting = ( state, settingName ) => {
	const unsaved = get( getUnsavedSettings( state ), settingName );

	// if the change exists in our unsaved queue, return it
	// otherwise try to get the saved value
	// fail to `null`
	return unsaved
		? unsaved
		: get( getUserSettings( state ), settingName, null );
}
```